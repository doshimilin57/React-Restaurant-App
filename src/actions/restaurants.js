import axios from 'axios';
import { SET_RESTAURANTS, SHOW_MORE_RESTAURANTS, SET_PAGE } from './types';
import { setRestaurantsCount } from './results';

// SET_RESTAURANTS
export const setRestaurants = (restaurants) => ({
  type: SET_RESTAURANTS,
  restaurants,
});

// SHOW_MORE_RESTAURANTS
export const setShowMoreRestaurants = (restaurants) => ({
  type: SHOW_MORE_RESTAURANTS,
  restaurants,
});

// SHOW_MORE_RESTAURANTS
export const setPage = (page) => ({
  type: SET_PAGE,
  page,
});

export const startSearchRestaurants = (city = '') => {
  return async (dispatch) => {
    try {
      const res = await axios.get(
        `http://opentable.herokuapp.com/api/restaurants?city=${city}&per_page=25&page=1`
      );
      dispatch(setRestaurants(res.data.restaurants));
      dispatch(setRestaurantsCount(res.data.total_entries));
    } catch (e) {
      console.log(e, e.response);
    }
  };
};

export const startShowMoreRestaurants = (city = '') => {
  return async (dispatch, getState) => {
    try {
      const page = getState().results.page || 1;
      const res = await axios.get(
        `http://opentable.herokuapp.com/api/restaurants?city=${city}&per_page=25&page=${
          page + 1
        }`
      );
      dispatch(setShowMoreRestaurants(res.data.restaurants));
      dispatch(setPage(page + 1));
    } catch (e) {
      console.log(e, e.response);
    }
  };
};
