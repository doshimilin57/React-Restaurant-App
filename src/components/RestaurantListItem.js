import React from 'react';
import numeral from 'numeral';

const RestaurantListItem = ({
  id,
  name,
  price,
  address,
  city,
  state,
  country,
  postal_code,
}) => (
  <div className="list-item">
    <div>
      <h3 className="list-item__title">{name}</h3>
      <span className="list-item__sub-title">
        {address}
        {', '}
        {city}
        {', '}
        {state}
        {', '}
        {country}
        {', '}
        {postal_code}
      </span>
    </div>
    <h3 className="list-item__data">{numeral(price).format('$0,0.00')}</h3>
  </div>
);

export default RestaurantListItem;
