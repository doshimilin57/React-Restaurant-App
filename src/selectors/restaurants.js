import moment from 'moment';

// Get visible restaurants

export default (restaurants, { text }) => {
  return restaurants.filter((restaurant) => {
    const textMatch = restaurant.name
      .toLowerCase()
      .includes(text.toLowerCase());
    const addressMatch = restaurant.address
      .toLowerCase()
      .includes(text.toLowerCase());
    const areaMatch = restaurant.area
      .toLowerCase()
      .includes(text.toLowerCase());
    const cityMatch = restaurant.city
      .toLowerCase()
      .includes(text.toLowerCase());
    const stateMatch = restaurant.state
      .toLowerCase()
      .includes(text.toLowerCase());
    const postalCodeMatch = restaurant.postal_code
      .toLowerCase()
      .includes(text.toLowerCase());
    return (
      textMatch ||
      addressMatch ||
      areaMatch ||
      cityMatch ||
      stateMatch ||
      postalCodeMatch
    );
  });
};
