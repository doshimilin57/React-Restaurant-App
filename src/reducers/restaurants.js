// Restaurants Reducer

import { SET_RESTAURANTS, SHOW_MORE_RESTAURANTS } from '../actions/types';

const restaurantsReducerDefaultState = [];

export default (state = restaurantsReducerDefaultState, action) => {
  switch (action.type) {
    case SET_RESTAURANTS:
      return action.restaurants;
    case SHOW_MORE_RESTAURANTS:
      return [...state, ...action.restaurants];
    default:
      return state;
  }
};
