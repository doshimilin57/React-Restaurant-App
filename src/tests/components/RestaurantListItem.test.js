import React from 'react';
import { shallow } from 'enzyme';
import RestaurantListItem from '../../components/RestaurantListItem';
import restaurants from '../fixtures/restaurants';

test('should render books list item with book', () => {
  const wrapper = shallow(<RestaurantListItem {...restaurants[0]} />);
  expect(wrapper).toMatchSnapshot();
});
