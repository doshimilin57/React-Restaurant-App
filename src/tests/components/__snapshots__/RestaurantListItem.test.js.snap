// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render books list item with book 1`] = `
<div
  className="list-item"
>
  <div>
    <h3
      className="list-item__title"
    >
      Edulis Restaurant
    </h3>
    <span
      className="list-item__sub-title"
    >
      169 Niagara Street
      , 
      Toronto
      , 
      ON
      , 
      CA
      , 
      M5V 1C9
    </span>
  </div>
  <h3
    className="list-item__data"
  >
    $3.00
  </h3>
</div>
`;
